packages/rs-dpp/src/identity/state_transition/identity_credit_withdrawal_transition/apply_identity_credit_withdrawal_transition_factory.rs

```diff
@@ -34,37 +40,73 @@ where
         &self,
         state_transition: &IdentityCreditWithdrawalTransition,
     ) -> Result<()> {
-        let latest_withdrawal_index = self
+        let data_contract_id = Identifier::from_string(
+            &withdrawals_contract::system_ids().contract_id,
+            Encoding::Base58,
+        )?;
+        let data_contract_owner_id = Identifier::from_string(
+            &withdrawals_contract::system_ids().owner_id,
+            Encoding::Base58,
+        )?;
+
+        let maybe_withdrawals_data_contract: Option<DataContract> = self
             .state_repository
-            .fetch_latest_withdrawal_transaction_index()
-            .await?;
+            .fetch_data_contract(&data_contract_id, state_transition.get_execution_context())
+            .await?
+            .map(TryInto::try_into)
+            .transpose()
+            .map_err(Into::into)?;
 
-        let output_script: Script = state_transition.output_script.deref().clone();
+        let withdrawals_data_contract = maybe_withdrawals_data_contract
+            .ok_or_else(|| anyhow!("Withdrawals data contract not found"))?;
 
-        let tx_out = TxOut {
-            value: convert_credits_to_satoshi(state_transition.amount),
-            script_pubkey: output_script,
-        };
+        let latest_platform_block_header: JsonValue = self
+            .state_repository
+            .fetch_latest_platform_block_header()
+            .await?;
 
-        let withdrawal_transaction = AssetUnlockBaseTransactionInfo {
-            version: 1,
-            lock_time: 0,
-            output: vec![tx_out],
-            base_payload: AssetUnlockBasePayload {
-                version: 1,
-                index: latest_withdrawal_index + 1,
-                fee: state_transition.core_fee,
-            },
-        };
+        let document_type = String::from(withdrawals_contract::types::WITHDRAWAL);
+        let document_entropy = generate()?;
+        let document_created_at_millis = latest_platform_block_header
+            .get(PLATFORM_BLOCK_HEADER_TIME_PROPERTY)
+            .ok_or_else(|| anyhow!("time property is not set in block header"))?
+            .get_i64(PLATFORM_BLOCK_HEADER_TIME_SECONDS_PROPERTY)?
+            * 1000;
+
+        let document_data = json!({
+            "amount": state_transition.amount,
+            "coreFeePerByte": state_transition.core_fee_per_byte,
+            "pooling": Pooling::Never,
+            "outputScript": state_transition.output_script.as_bytes(),
+            "status": withdrawals_contract::statuses::QUEUED,
+        });
 
-        let mut transaction_buffer: Vec<u8> = vec![];
+        let document_id_bytes: [u8; 32] = state_transition
```

Comment:
You should construct document ID from contract ID, owner ID and entropy such as it's done in the document factory. State transition's hash can be used as an entry.